app/scenes/Document/components/CommentSortMenu.tsx

```diff
@@ -0,0 +1,61 @@
+import { ManualSortIcon } from "outline-icons";
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { MenuButton, useMenuState } from "reakit";
+import Button from "~/components/Button";
+import ContextMenu from "~/components/ContextMenu";
+import Template from "~/components/ContextMenu/Template";
+import { CommentSortType, MenuItem } from "~/types";
+
+type Props = {
+  value: CommentSortType;
+  onSelect: (type: CommentSortType) => void;
+};
+
+const CommentSortMenu: React.FC<Props> = ({ value, onSelect }) => {
+  const { t } = useTranslation();
+  const menu = useMenuState({ modal: true });
+
+  const items: MenuItem[] = React.useMemo(
+    () => [
+      {
+        type: "button",
+        title: t("Chronological sort"),
+        visible: true,
+        selected: value === CommentSortType.Chrono,
+        onClick: () => {
+          if (value !== CommentSortType.Chrono) {
+            onSelect(CommentSortType.Chrono);
+          }
+        },
+      },
+      {
+        type: "button",
+        title: t("Positional sort"),
+        visible: true,
+        selected: value === CommentSortType.Position,
+        onClick: () => {
+          if (value !== CommentSortType.Position) {
+            onSelect(CommentSortType.Position);
+          }
+        },
+      },
+    ],
+    [t, value, onSelect]
+  );
+
+  return (
+    <>
+      <MenuButton {...menu} aria-label={t("Sort comments")}>
+        {(props) => (
+          <Button {...props} neutral borderOnHover icon={<ManualSortIcon />} />
```

Comment:
Tippy doesn't play well with the menu button here. We'd have to maintain state and add some complex handlers (quite tricky, I think) to enable tooltip. Do we want to do it?

Reply:
No, it's okay – I think we need to change this somehow anyway – having three icons in a row is too much, especially as this sort option is something you'll change once and then likely never change again.

I'll have a think and get back to you

Reply:
Probably better off moving this to preferences page?

Reply:
It's nice to directly see the effect it has, on preferences it's not as clear. I think the answer is to combine the resolved button and this into a "Display options" type of menu – I can probably make that change in the next few days, just need to iterate a bit on how it should be.

Reply:
Got it - I thought about this too, just wasn't sure if it's ideal to move the resolved comments toggle inside a menu. But I guess, we don't much option here..

Lmk in case you want me to implement it once the design is ready.